### PR TITLE
refactor!: group variables for better overview

### PR DESCRIPTION
**Pull request by kayman-mk**
## Description

Groups variables into objects to
- reduce the number of variables (currently 118)
- to gain a better overview of all configuration settings

Creates new groups of variables:
- `runner_manager` in case it configures the "main" process which sets the defaults for all runners
- `runner` in case it configures the runner created by the runner manager
- `runner_worker` in case it configures the docker/docker+machine or leave it as it is, if it is a global scope, e.g. common tags, the environment, ...

## Migrations required

Yes and a script is provided to do that. It covers 98% of all migrations (see migrations/migrate-to-7-0-0.sh)

## Verification

Please mention the examples you have verified.

### CODE REVIEW

**Comment by kayman-mk** [comment]
Branch is based on #757 Commits beginning from here are the relevant ones.

**Comment by kayman-mk** [comment]
@npalm This one is ready now. It rewrites the complete `variables.tf` and restructures it. Could you please do a review here? Extensive tests still to be done. But I will do it on the target branch `refactor-variables` this PR points to. There are some other major changes pending.

#723

File: examples/runner-certificates/README.md [hunk 1]
```diff
@@ -34,7 +34,7 @@ Create a PEM-encoded `.crt` file containing the public certificate of your Gitla
 module {
   ...
   # Public cert of my companys gitlab instance
-  runners_gitlab_certificate = file("${path.module}/my_gitlab_instance_cert.crt")
+  agent_gitlab_certificate = file("${path.module}/my_gitlab_instance_cert.crt")
```
Comment (npalm) [inline examples/runner-certificates/README.md:37]: Should this not be runner_gitlab.... ?

**Review by npalm — CHANGES_REQUESTED** [review]
Looks in general good. Will run a check also locally.

**Comment by npalm** [comment]
Scopting with refactor will not create a new release, and also will not be mentioned int he release note as far I know. Maybe better to just scope it as feat.

**Comment by kayman-mk** [comment]
The `!` should be the solution as it creates a major release. Let me quickly check this.

**Comment by kayman-mk** [comment]
@npalm The only problem I see, is, how to manually update the release docs. Tried some changes here and there, but Release-Please overwrites them. Any idea how to change the docs manually?

I would like to have a reference to the migration script at a prominent place.

File: examples/runner-certificates/README.md (hunk 1 above)
Comment (kayman-mk) [reply examples/runner-certificates/README.md]: Thanks for pointing at this. I took the chance to validate all `README.md` files and found some other problems too. Fixed.

**Comment by kayman-mk** [comment]
@npalm Rebasing is a pain here. Can we merge to target branch `kayma/group-variables` to start testing the v7 release?

**Comment by kayman-mk** [comment]
Merged into the pre-release branch for testing.

**Pull request merged**